master/static/srv/get_aggregated_allocation.sql

```diff
@@ -1,65 +1,46 @@
-WITH const AS (
-    SELECT daterange($1::date, $2::date, '[]') AS period
-),
+WITH date_series AS (
+  SELECT generate_series(
+    GREATEST($1::date, (SELECT min(date) FROM resource_aggregates)),
+    LEAST($2::date, (SELECT max(date) FROM resource_aggregates)),
+    '1 day'::interval
+  ) AS period_start
+)
+SELECT
+  to_char(ds.period_start, 'YYYY-MM-DD') AS period_start,
+  'RESOURCE_ALLOCATION_AGGREGATION_PERIOD_DAILY' AS period,
 
-days AS (
-    SELECT
-        resource_aggregates.date::date AS period_start,
-        aggregation_type,
-        resource_aggregates.aggregation_key,
-        seconds
-    FROM
-        resource_aggregates,
-        const
-    WHERE
-        -- `@>` determines whether the range contains the time.
-        const.period @> resource_aggregates.date
-),
+  -- We divide by count since we have multiple rows with the same total.
+  (SUM(ra_total.seconds) FILTER (WHERE ra_total.aggregation_key IS NOT NULL) /
+    COUNT(*) FILTER (WHERE ra_total.aggregation_key IS NOT NULL)) AS seconds,
```

Comment:
question: are the rows always guaranteed to have the same second values?

Reply:
Yeah they always will, but I think there was some awkwardness in the LEFT JOIN method, I changed my mind and decided to rewrite the query and use the same method as the original but making sure index scans happen